### DISCUSSION TITLE
coder AI/ML related things

### DISCUSSION THREAD

**Discussion by da00891219** [coder/coder#8027] [Announcements]
What do we have in AI space in coder? 
For example to improve developer productivity, what tools do we have?
Do we have anything GitHub Copilot or similar products?

**Comment by matifali** [comment]
Coder allows you to connect to your workspaces using VSCode and JetBrains IDEs. Both of them have official GitHub copilot support

**Reply by matifali** [reply]
If you are looking for templates. I used that one account for my own research
https://github.com/matifali/coder-templates

**Reply by anilkintala123** [reply]
Hi @matifali , thanks for the response. 

Did you actually use the copilot with vscode in coder worksapce ?

The actual reason for this thread is because "coder based vscode does not support copilot". This has been confirmed by the coder support. 
Here is the history of our research in this regard. 
- coder workspaces implements code-server (https://github.com/coder/code-server) for its browser based vscode. This coder-server is a fork of opensource vscode and is maintained by the coder.com team itself. 
- therefore, the code-server based vscode implementation is different from microsoft offered vscode https://code.visualstudio.com/
- They are two different things, both originating from the opensource vscode. 

Now coming to the actual point of copilot support in coder vscode 
- copilot, which is a product of github, which inturn is owned by microsoft, only supports MicroSoft supported VSCODE ( https://code.visualstudio.com/). 
- copilot It does not support the code-server based vscode , which is available in coder workspaces. 

- This information is confirmed by the coder team. 

Because of this confirmation, we are checking with the coder team, if they have any AI based plugins that they are working on , that can support code-server based VSCODE.

**Reply by matifali** [reply]
Hi, @anilkintala123 You are mixing two different things.

1. [`code-server`](https://github.com/coder/code-server) is a web-based IDE that allows you to access the open source VS Code in your browser. You certainly can't install Microsoft-licensed extensions here, e.g., Copilot.
2. Coder provides [`vscode-coder`](https://github.com/coder/vscode-coder) and [`jetbrains-coder`](https://github.com/coder/jetbrains-coder) official plugins, which allows you to connect to any Coder workspace from your locally installed VS Code and JetBrains Gateway respectively. So can legally install GitHub Copilot with both of these options.

I hope you understand your confusion. If you want a demo for Coder Enterprise, you can contact [here](https://coder.com/demo).

**Reply by anilkintala123** [reply]
Thank you. @matifali  . 
I am still trying to wrap my head around this.  I didnt realize there is a plugin **vscode-coder** exists that can connect a locally installed vscode to a coder workspace. 

so, please help me out with the flow here. 
- i install vscode ide (https://code.visualstudio.com/) on my laptop (this is by microsoft). which allows me to install copilot 
- then i install **vscode-coder** plugin on my laptop IDE which lets me connect to a remote coder workspace 

That means, i am not using the vscode IDE which comes as part of coder workspace. Is that correct ?

**Reply by matifali** [reply]
The easiest way to start is using the VSCode Desktop button on the dashboard. 
By clicking on this button it should automatically launch your locally installed vs code. When you are successfully connected to your workspace you can install Microsoft copilot extension and use it.

**Reply by matifali** [reply]
@anilkintala123 I hope you have got your answer. Should we mark this discussion as answered?

**Comment by anilkintala123** [comment]
Yes please
